common/src/main/kotlin/entity/DiscordMessage.kt

```diff
@@ -719,42 +719,77 @@ data class AllRemovedMessageReactions(
 )
 
 @Serializable(with = MessageType.MessageTypeSerializer::class)
-enum class MessageType(val code: Int) {
+sealed class MessageType(val code: Int) {
     /** The default code for unknown values. */
-    Unknown(Int.MIN_VALUE),
-    Default(0),
-    RecipientAdd(1),
-    RecipientRemove(2),
-    Call(3),
-    ChannelNameChange(4),
-    ChannelIconChange(5),
-    ChannelPinnedMessage(6),
-    GuildMemberJoin(7),
-    UserPremiumGuildSubscription(8),
-    UserPremiumGuildSubscriptionTierOne(9),
-    UserPremiumGuildSubscriptionTwo(10),
-    UserPremiumGuildSubscriptionThree(11),
-    ChannelFollowAdd(12),
-    GuildDiscoveryDisqualified(14),
+    class Unknown(code: Int) : MessageType(code)
+    object Default : MessageType(0)
+    object RecipientAdd : MessageType(1)
+    object RecipientRemove : MessageType(2)
+    object Call : MessageType(3)
+    object ChannelNameChange : MessageType(4)
+    object ChannelIconChange : MessageType(5)
+    object ChannelPinnedMessage : MessageType(6)
+    object GuildMemberJoin : MessageType(7)
+    object UserPremiumGuildSubscription : MessageType(8)
+    object UserPremiumGuildSubscriptionTierOne : MessageType(9)
+    object UserPremiumGuildSubscriptionTwo : MessageType(10)
+    object UserPremiumGuildSubscriptionThree : MessageType(11)
+    object ChannelFollowAdd : MessageType(12)
+    object GuildDiscoveryDisqualified : MessageType(14)
 
     @Suppress("SpellCheckingInspection")
-    GuildDiscoveryRequalified(15),
-    Reply(19);
+    object GuildDiscoveryRequalified : MessageType(15)
+    object GuildDiscoveryGracePeriodInitialWarning : MessageType(16)
+    object GuildDiscoveryGracePeriodFinalWarning : MessageType(17)
+    object ThreadCreated : MessageType(18)
+    object Reply : MessageType(19)
+    object ApplicationCommand : MessageType(20)
+    object ThreadStarterMessage : MessageType(21)
+    object GuildInviteReminder : MessageType(22)
 
-    companion object MessageTypeSerializer : KSerializer<MessageType> {
+    object MessageTypeSerializer : KSerializer<MessageType> {
```

Comment:
While we're at it, make this internal, users have no need for it

```suggestion
    internal object MessageTypeSerializer : KSerializer<MessageType> {
```